autolift-scalaz/src/main/scala/autolift/scalaz/package.scala

```diff
@@ -0,0 +1,12 @@
+package autolift
+
+import _root_.scalaz.Functor
+import _root_.scalaz.Scalaz._
+
+package object scalaz extends autolift.AutoLifts {
```

Comment:
This is an interesting thought! I'd love to see if you could get it compiling. 

I've tried using type params to implement this type of thing before but ran into issues of implicit search failures, i.e.
```scala
trait Thingie[Functor[_[_]]{
```
there's a ton of repetition in the code bases specifically because of the implicit parameters (as well as differences in implementations, i.e. `Applicative`)